

This notebook contains an example for teaching.


# A Simple Case Study using Wage Data from 2015 - proceeding

So far we considered many machine learning method, e.g Lasso and Random Forests, to build a predictive model. In this lab, we extend our toolbox by predicting wages by a neural network.

## Data preparation

Again, we consider data from the U.S. March Supplement of the Current Population Survey (CPS) in 2015.



This notebook contains an example for teaching.


# A Simple Case Study using Wage Data from 2015 - proceeding

So far we considered many machine learning method, e.g Lasso and Random Forests, to build a predictive model. In this lab, we extend our toolbox by predicting wages by a neural network.

## Data preparation

Again, we consider data from the U.S. March Supplement of the Current Population Survey (CPS) in 2015.

In [1]:
# Import relevant packages
import pandas as pd
import numpy as np
import pyreadr
from sklearn import preprocessing
import patsy

from numpy import loadtxt
from keras.models import Sequential
from keras.layers import Dense

In [2]:
rdata_read = pyreadr.read_r("../data/wage2015_subsample_inference.Rdata")
data = rdata_read[ 'data' ]
n = data.shape[0]

In [3]:
# Import relevant packages for splitting data
import random
import math
# Set Seed
# to make the results replicable (generating random numbers)
np.random.seed(0)
random = np.random.randint(0, data.shape[0], size=math.floor(data.shape[0]))
data["random"] = random
random    # the array does not change 

array([2732, 2607, 1653, ..., 4184, 2349, 3462])

In [4]:
data_2 = data.sort_values(by=['random'])
data_2

wage     lwage  sex  shs  hsg  scl  clg   ad   mw   so  ...  \
rownames                                                               ...   
2223      26.442308  3.274965  1.0  0.0  1.0  0.0  0.0  0.0  0.0  0.0  ...   
3467      19.230769  2.956512  0.0  0.0  0.0  1.0  0.0  0.0  0.0  0.0  ...   
13501     48.076923  3.872802  1.0  0.0  0.0  0.0  0.0  1.0  1.0  0.0  ...   
15588     12.019231  2.486508  0.0  0.0  1.0  0.0  0.0  0.0  0.0  1.0  ...   
16049     39.903846  3.686473  1.0  0.0  0.0  0.0  0.0  1.0  0.0  1.0  ...   
...             ...       ...  ...  ...  ...  ...  ...  ...  ...  ...  ...   
27533     21.634615  3.074295  1.0  0.0  0.0  0.0  1.0  0.0  0.0  0.0  ...   
7218      13.461538  2.599837  0.0  0.0  1.0  0.0  0.0  0.0  1.0  0.0  ...   
7204      27.403846  3.310683  1.0  0.0  0.0  0.0  0.0  1.0  1.0  0.0  ...   
1380      16.695804  2.815157  1.0  0.0  0.0  1.0  0.0  0.0  0.0  0.0  ...   
10451     45.528846  3.818346  1.0  0.0  0.0  0.0  1.0  0.0  1.0  0.0  ...   

           ne  exp1     exp2       exp3        exp4   occ occ2   ind ind2  \
rownames                                                                    
2223      1.0  29.0   8.4100  24.389000   70.728100   340    1  8660   20   
3467      1.0  33.5  11.2225  37.595375  125.944506  9620   22  1870    5   
13501     0.0   2.0   0.0400   0.008000    0.001600  3060   10  8190   18   
15588     0.0  29.0   8.4100  24.389000   70.728100  6440   19   770    4   
16049     0.0  12.0   1.4400   1.728000    2.073600  1820    5  7860   17   
...       ...   ...      ...        ...         ...   ...  ...   ...  ...   
27533     0.0  10.0   1.0000   1.000000    1.000000  1860    5  7870   17   
7218      0.0  28.0   7.8400  21.952000   61.465600  4220   14  1170    5   
7204      0.0   4.0   0.1600   0.064000    0.025600  1760    5  7460   14   
1380      1.0  28.0   7.8400  21.952000   61.465600  2000    6  9570   22   
10451     0.0  11.0   1.2100   1.331000    1.464100  1107    3  6990   12   

         random  
rownames         
2223          0  
3467          0  
13501         0  
15588         2  
16049         2  
...         ...  
27533      5146  
7218       5146  
7204       5146  
1380       5149  
10451      5149  

[5150 rows x 21 columns]

In [5]:
# Create training and testing sample 
data_train = data_2[ : math.floor(n*3/4)]    # training sample
data_test =  data_2[ math.floor(n*3/4) : ]   # testing sample
print(data_train.shape)
print(data_test.shape)

(3862, 21)
(1288, 21)


In [6]:
data_train = data_train.iloc[:, 0:16]
data_test = data_test.iloc[:, 0:16] 
data_test

wage     lwage  sex  shs  hsg  scl  clg   ad   mw   so   we  \
rownames                                                                     
9210      29.807692  3.394766  1.0  0.0  1.0  0.0  0.0  0.0  1.0  0.0  0.0   
16484     43.269231  3.767442  1.0  0.0  0.0  0.0  1.0  0.0  0.0  1.0  0.0   
16448     24.038462  3.179655  1.0  0.0  0.0  0.0  1.0  0.0  0.0  1.0  0.0   
27392     10.097115  2.312250  0.0  0.0  1.0  0.0  0.0  0.0  0.0  0.0  1.0   
11596      8.653846  2.158004  1.0  0.0  1.0  0.0  0.0  0.0  1.0  0.0  0.0   
...             ...       ...  ...  ...  ...  ...  ...  ...  ...  ...  ...   
27533     21.634615  3.074295  1.0  0.0  0.0  0.0  1.0  0.0  0.0  0.0  1.0   
7218      13.461538  2.599837  0.0  0.0  1.0  0.0  0.0  0.0  1.0  0.0  0.0   
7204      27.403846  3.310683  1.0  0.0  0.0  0.0  0.0  1.0  1.0  0.0  0.0   
1380      16.695804  2.815157  1.0  0.0  0.0  1.0  0.0  0.0  0.0  0.0  0.0   
10451     45.528846  3.818346  1.0  0.0  0.0  0.0  1.0  0.0  1.0  0.0  0.0   

           ne  exp1  exp2    exp3     exp4  
rownames                                    
9210      0.0  22.0  4.84  10.648  23.4256  
16484     0.0  17.0  2.89   4.913   8.3521  
16448     0.0  11.0  1.21   1.331   1.4641  
27392     0.0  30.0  9.00  27.000  81.0000  
11596     0.0   6.0  0.36   0.216   0.1296  
...       ...   ...   ...     ...      ...  
27533     0.0  10.0  1.00   1.000   1.0000  
7218      0.0  28.0  7.84  21.952  61.4656  
7204      0.0   4.0  0.16   0.064   0.0256  
1380      1.0  28.0  7.84  21.952  61.4656  
10451     0.0  11.0  1.21   1.331   1.4641  

[1288 rows x 16 columns]

In [7]:
# normalize the data
from sklearn.preprocessing import MinMaxScaler

scaler =  MinMaxScaler().fit(data_train)
scaler =  MinMaxScaler().fit(data_test)

# scaler = preprocessing.StandardScaler().fit(data_train)
# scaler = preprocessing.StandardScaler().fit(data_test)

data_train_scaled = scaler.transform(data_train)
data_test_scaled = scaler.transform(data_test)


In [8]:
columns = list(data_train)

In [9]:
data_train_scaled = pd.DataFrame(data_train_scaled, columns = columns)
data_test_scaled = pd.DataFrame(data_test_scaled, columns = columns)
data_test_scaled

wage     lwage  sex  shs  hsg  scl  clg   ad   mw   so   we   ne  \
0     0.054875  0.447819  1.0  0.0  1.0  0.0  0.0  0.0  1.0  0.0  0.0  0.0   
1     0.082510  0.521313  1.0  0.0  0.0  0.0  1.0  0.0  0.0  1.0  0.0  0.0   
2     0.043032  0.405398  1.0  0.0  0.0  0.0  1.0  0.0  0.0  1.0  0.0  0.0   
3     0.014412  0.234342  0.0  0.0  1.0  0.0  0.0  0.0  0.0  0.0  1.0  0.0   
4     0.011449  0.203924  1.0  0.0  1.0  0.0  0.0  0.0  1.0  0.0  0.0  0.0   
...        ...       ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...   
1283  0.038097  0.384621  1.0  0.0  0.0  0.0  1.0  0.0  0.0  0.0  1.0  0.0   
1284  0.021318  0.291056  0.0  0.0  1.0  0.0  0.0  0.0  1.0  0.0  0.0  0.0   
1285  0.049940  0.431238  1.0  0.0  0.0  0.0  0.0  1.0  1.0  0.0  0.0  0.0   
1286  0.027958  0.333518  1.0  0.0  0.0  1.0  0.0  0.0  0.0  0.0  0.0  1.0   
1287  0.087149  0.531351  1.0  0.0  0.0  0.0  1.0  0.0  1.0  0.0  0.0  0.0   

          exp1      exp2      exp3      exp4  
0     0.500000  0.250000  0.125000  0.062500  
1     0.386364  0.149277  0.057675  0.022284  
2     0.250000  0.062500  0.015625  0.003906  
3     0.681818  0.464876  0.316961  0.216110  
4     0.136364  0.018595  0.002536  0.000346  
...        ...       ...       ...       ...  
1283  0.227273  0.051653  0.011739  0.002668  
1284  0.636364  0.404959  0.257701  0.163992  
1285  0.090909  0.008264  0.000751  0.000068  
1286  0.636364  0.404959  0.257701  0.163992  
1287  0.250000  0.062500  0.015625  0.003906  

[1288 rows x 16 columns]

Then, we construct the inputs for our network.

In [10]:
formula_basic = "lwage ~ sex + exp1 + shs + hsg+ scl + clg + mw + so + we"
Y_train, model_X_basic_train = patsy.dmatrices(formula_basic, data_train_scaled, return_type='dataframe')
Y_test, model_X_basic_test = patsy.dmatrices(formula_basic, data_test_scaled, return_type='dataframe')

### Neural Networks

First, we need to determine the structure of our network. We are using the R/python package *keras* to build a simple sequential neural network with three dense layers.

In [11]:
model_X_basic_train.shape[1]

10

In [12]:
# define the keras model
model = Sequential()
model.add(Dense(20, input_dim = model_X_basic_train.shape[1], activation = 'relu'))
model.add(Dense(10, activation = 'relu'))
# model.add(Dense(5, activation = 'relu'))

model.add(Dense(1))

In [13]:
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers


In [14]:
# compile the keras model
opt = keras.optimizers.Adam(learning_rate=0.005)
mse = tf.keras.losses.MeanSquaredError()
mae = tf.keras.metrics.MeanAbsoluteError(name="mean_absolute_error", dtype=None)

Let us have a look at the structure of our network in detail.

In [15]:
model.compile(loss=mse, optimizer= opt , metrics=mae)
model.summary(line_length=None, positions=None, print_fn=None)

Model: "sequential"


_________________________________________________________________


 Layer (type)                Output Shape              Param #   


 dense (Dense)               (None, 20)                220       


 dense_1 (Dense)             (None, 10)                210       


 dense_2 (Dense)             (None, 1)                 11        


Total params: 441


Trainable params: 441


Non-trainable params: 0


_________________________________________________________________


It is worth to notice that we have in total $441$ trainable parameters.

Now, let us train the network. Note that this takes some computation time. Thus, we are using gpu to speed up. The exact speed-up varies based on a number of factors including model architecture, batch-size, input pipeline complexity, etc.

In [16]:
# fit the keras model on the dataset
num_epochs = 1000

Check this [link](https://stats.stackexchange.com/questions/153531/what-is-batch-size-in-neural-network), to understand batch_size argument

In [17]:
# fit the keras model on the dataset
model.fit(model_X_basic_train, Y_train, epochs=150, batch_size=10)

Epoch 1/150


  1/387 [..............................] - ETA: 6:00 - loss: 0.0337 - mean_absolute_error: 0.1369

 24/387 [>.............................] - ETA: 0s - loss: 0.0155 - mean_absolute_error: 0.0990  

 50/387 [==>...........................] - ETA: 0s - loss: 0.0151 - mean_absolute_error: 0.0956

 70/387 [====>.........................] - ETA: 0s - loss: 0.0140 - mean_absolute_error: 0.0915

 92/387 [======>.......................] - ETA: 0s - loss: 0.0145 - mean_absolute_error: 0.0927

118/387 [========>.....................] - ETA: 0s - loss: 0.0140 - mean_absolute_error: 0.0914

143/387 [==========>...................] - ETA: 0s - loss: 0.0135 - mean_absolute_error: 0.0895

168/387 [============>.................] - ETA: 0s - loss: 0.0134 - mean_absolute_error: 0.0887

190/387 [=============>................] - ETA: 0s - loss: 0.0131 - mean_absolute_error: 0.0878

206/387 [==============>...............] - ETA: 0s - loss: 0.0128 - mean_absolute_error: 0.0872

231/387 [================>.............] - ETA: 0s - loss: 0.0129 - mean_absolute_error: 0.0872

254/387 [==================>...........] - ETA: 0s - loss: 0.0128 - mean_absolute_error: 0.0872

280/387 [====================>.........] - ETA: 0s - loss: 0.0130 - mean_absolute_error: 0.0876

305/387 [======================>.......] - ETA: 0s - loss: 0.0130 - mean_absolute_error: 0.0873

328/387 [========================>.....] - ETA: 0s - loss: 0.0129 - mean_absolute_error: 0.0872

349/387 [==========================>...] - ETA: 0s - loss: 0.0127 - mean_absolute_error: 0.0864

368/387 [===========================>..] - ETA: 0s - loss: 0.0126 - mean_absolute_error: 0.0862

387/387 [==============================] - ETA: 0s - loss: 0.0124 - mean_absolute_error: 0.0856

387/387 [==============================] - 2s 2ms/step - loss: 0.0124 - mean_absolute_error: 0.0856


Epoch 2/150


  1/387 [..............................] - ETA: 1s - loss: 0.0119 - mean_absolute_error: 0.1000

 20/387 [>.............................] - ETA: 1s - loss: 0.0131 - mean_absolute_error: 0.0905

 35/387 [=>............................] - ETA: 1s - loss: 0.0122 - mean_absolute_error: 0.0869

 54/387 [===>..........................] - ETA: 0s - loss: 0.0126 - mean_absolute_error: 0.0870

 73/387 [====>.........................] - ETA: 0s - loss: 0.0122 - mean_absolute_error: 0.0854

 94/387 [======>.......................] - ETA: 0s - loss: 0.0120 - mean_absolute_error: 0.0847

116/387 [=======>......................] - ETA: 0s - loss: 0.0117 - mean_absolute_error: 0.0837

135/387 [=========>....................] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0825

157/387 [===========>..................] - ETA: 0s - loss: 0.0112 - mean_absolute_error: 0.0826

178/387 [============>.................] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0819

197/387 [==============>...............] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0821

218/387 [===============>..............] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0818

233/387 [=================>............] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0814

254/387 [==================>...........] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0817

273/387 [====================>.........] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0818

295/387 [=====================>........] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0817

317/387 [=======================>......] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0814

340/387 [=========================>....] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0815

361/387 [==========================>...] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0818

383/387 [============================>.] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0815

387/387 [==============================] - 1s 3ms/step - loss: 0.0113 - mean_absolute_error: 0.0815


Epoch 3/150


  1/387 [..............................] - ETA: 1s - loss: 0.0125 - mean_absolute_error: 0.0940

 21/387 [>.............................] - ETA: 0s - loss: 0.0126 - mean_absolute_error: 0.0886

 42/387 [==>...........................] - ETA: 0s - loss: 0.0118 - mean_absolute_error: 0.0852

 60/387 [===>..........................] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0821

 82/387 [=====>........................] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0832

104/387 [=======>......................] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0829

126/387 [========>.....................] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0817

148/387 [==========>...................] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0813

166/387 [===========>..................] - ETA: 0s - loss: 0.0112 - mean_absolute_error: 0.0815

188/387 [=============>................] - ETA: 0s - loss: 0.0114 - mean_absolute_error: 0.0826

208/387 [===============>..............] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0823

229/387 [================>.............] - ETA: 0s - loss: 0.0112 - mean_absolute_error: 0.0823

247/387 [==================>...........] - ETA: 0s - loss: 0.0111 - mean_absolute_error: 0.0819

263/387 [===================>..........] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0822

281/387 [====================>.........] - ETA: 0s - loss: 0.0112 - mean_absolute_error: 0.0819

301/387 [======================>.......] - ETA: 0s - loss: 0.0112 - mean_absolute_error: 0.0819

322/387 [=======================>......] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0819

343/387 [=========================>....] - ETA: 0s - loss: 0.0111 - mean_absolute_error: 0.0816

363/387 [===========================>..] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0817

384/387 [============================>.] - ETA: 0s - loss: 0.0112 - mean_absolute_error: 0.0815

387/387 [==============================] - 1s 3ms/step - loss: 0.0112 - mean_absolute_error: 0.0816


Epoch 4/150


  1/387 [..............................] - ETA: 1s - loss: 0.0048 - mean_absolute_error: 0.0596

 21/387 [>.............................] - ETA: 0s - loss: 0.0083 - mean_absolute_error: 0.0714

 42/387 [==>...........................] - ETA: 0s - loss: 0.0099 - mean_absolute_error: 0.0755

 60/387 [===>..........................] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0774

 75/387 [====>.........................] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0781

 94/387 [======>.......................] - ETA: 0s - loss: 0.0102 - mean_absolute_error: 0.0777

116/387 [=======>......................] - ETA: 0s - loss: 0.0100 - mean_absolute_error: 0.0772

135/387 [=========>....................] - ETA: 0s - loss: 0.0101 - mean_absolute_error: 0.0777

158/387 [===========>..................] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0788

181/387 [=============>................] - ETA: 0s - loss: 0.0100 - mean_absolute_error: 0.0776

204/387 [==============>...............] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0786

226/387 [================>.............] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0782

251/387 [==================>...........] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0791

274/387 [====================>.........] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0797

287/387 [=====================>........] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0793

308/387 [======================>.......] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0797

328/387 [========================>.....] - ETA: 0s - loss: 0.0111 - mean_absolute_error: 0.0803

349/387 [==========================>...] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0796

371/387 [===========================>..] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0798

387/387 [==============================] - 1s 3ms/step - loss: 0.0108 - mean_absolute_error: 0.0796


Epoch 5/150


  1/387 [..............................] - ETA: 1s - loss: 0.0109 - mean_absolute_error: 0.0758

 21/387 [>.............................] - ETA: 0s - loss: 0.0094 - mean_absolute_error: 0.0751

 45/387 [==>...........................] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0807

 66/387 [====>.........................] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0785

 86/387 [=====>........................] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0792

101/387 [======>.......................] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0797

121/387 [========>.....................] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0807

141/387 [=========>....................] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0811

163/387 [===========>..................] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0801

183/387 [=============>................] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0801

200/387 [==============>...............] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0799

222/387 [================>.............] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0791

242/387 [=================>............] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0793

265/387 [===================>..........] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0799

284/387 [=====================>........] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0800

299/387 [======================>.......] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0795

319/387 [=======================>......] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0795

340/387 [=========================>....] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0797

363/387 [===========================>..] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0795

384/387 [============================>.] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0797

387/387 [==============================] - 1s 3ms/step - loss: 0.0108 - mean_absolute_error: 0.0796


Epoch 6/150


  1/387 [..............................] - ETA: 1s - loss: 0.0120 - mean_absolute_error: 0.0834

 20/387 [>.............................] - ETA: 1s - loss: 0.0097 - mean_absolute_error: 0.0761

 42/387 [==>...........................] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0776

 61/387 [===>..........................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0791

 81/387 [=====>........................] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0787

102/387 [======>.......................] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0797

112/387 [=======>......................] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0789

133/387 [=========>....................] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0787

155/387 [===========>..................] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0787

176/387 [============>.................] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0794

196/387 [==============>...............] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0796

214/387 [===============>..............] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0796

235/387 [=================>............] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0789

256/387 [==================>...........] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0789

276/387 [====================>.........] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0784

296/387 [=====================>........] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0789

313/387 [=======================>......] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0786

327/387 [========================>.....] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0788

356/387 [==========================>...] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0787

382/387 [============================>.] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0789

387/387 [==============================] - 1s 3ms/step - loss: 0.0107 - mean_absolute_error: 0.0787


Epoch 7/150


  1/387 [..............................] - ETA: 1s - loss: 0.0060 - mean_absolute_error: 0.0556

 29/387 [=>............................] - ETA: 0s - loss: 0.0094 - mean_absolute_error: 0.0750

 53/387 [===>..........................] - ETA: 0s - loss: 0.0102 - mean_absolute_error: 0.0787

 79/387 [=====>........................] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0801

103/387 [======>.......................] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0792

125/387 [========>.....................] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0786

151/387 [==========>...................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0786

173/387 [============>.................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0790

191/387 [=============>................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0790

214/387 [===============>..............] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0780

238/387 [=================>............] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0782

262/387 [===================>..........] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0789

289/387 [=====================>........] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0793

316/387 [=======================>......] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0791

338/387 [=========================>....] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0786

363/387 [===========================>..] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0788

387/387 [==============================] - 1s 2ms/step - loss: 0.0107 - mean_absolute_error: 0.0789


Epoch 8/150


  1/387 [..............................] - ETA: 1s - loss: 0.0082 - mean_absolute_error: 0.0809

 24/387 [>.............................] - ETA: 0s - loss: 0.0091 - mean_absolute_error: 0.0744

 45/387 [==>...........................] - ETA: 0s - loss: 0.0089 - mean_absolute_error: 0.0740

 71/387 [====>.........................] - ETA: 0s - loss: 0.0101 - mean_absolute_error: 0.0775

 96/387 [======>.......................] - ETA: 0s - loss: 0.0101 - mean_absolute_error: 0.0774

120/387 [========>.....................] - ETA: 0s - loss: 0.0100 - mean_absolute_error: 0.0772

143/387 [==========>...................] - ETA: 0s - loss: 0.0100 - mean_absolute_error: 0.0762

169/387 [============>.................] - ETA: 0s - loss: 0.0099 - mean_absolute_error: 0.0754

197/387 [==============>...............] - ETA: 0s - loss: 0.0099 - mean_absolute_error: 0.0759

223/387 [================>.............] - ETA: 0s - loss: 0.0100 - mean_absolute_error: 0.0766

248/387 [==================>...........] - ETA: 0s - loss: 0.0101 - mean_absolute_error: 0.0771

274/387 [====================>.........] - ETA: 0s - loss: 0.0102 - mean_absolute_error: 0.0775

292/387 [=====================>........] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0782

316/387 [=======================>......] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0779

343/387 [=========================>....] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0780

369/387 [===========================>..] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0782

387/387 [==============================] - 1s 2ms/step - loss: 0.0105 - mean_absolute_error: 0.0785


Epoch 9/150


  1/387 [..............................] - ETA: 1s - loss: 0.0079 - mean_absolute_error: 0.0744

 26/387 [=>............................] - ETA: 0s - loss: 0.0117 - mean_absolute_error: 0.0809

 48/387 [==>...........................] - ETA: 0s - loss: 0.0117 - mean_absolute_error: 0.0821

 74/387 [====>.........................] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0810

 93/387 [======>.......................] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0800

117/387 [========>.....................] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0794

125/387 [========>.....................] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0802

137/387 [=========>....................] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0797

159/387 [===========>..................] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0799

175/387 [============>.................] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0792

196/387 [==============>...............] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0786

218/387 [===============>..............] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0787

236/387 [=================>............] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0791

252/387 [==================>...........] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0801

271/387 [====================>.........] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0804

289/387 [=====================>........] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0798

308/387 [======================>.......] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0794

322/387 [=======================>......] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0794

342/387 [=========================>....] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0796

361/387 [==========================>...] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0788

380/387 [============================>.] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0790

387/387 [==============================] - 1s 3ms/step - loss: 0.0107 - mean_absolute_error: 0.0788


Epoch 10/150


  1/387 [..............................] - ETA: 1s - loss: 0.0154 - mean_absolute_error: 0.1087

 19/387 [>.............................] - ETA: 1s - loss: 0.0098 - mean_absolute_error: 0.0759

 39/387 [==>...........................] - ETA: 0s - loss: 0.0094 - mean_absolute_error: 0.0760

 58/387 [===>..........................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0787

 81/387 [=====>........................] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0788

100/387 [======>.......................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0793

115/387 [=======>......................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0797

130/387 [=========>....................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0792

150/387 [==========>...................] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0792

169/387 [============>.................] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0786

189/387 [=============>................] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0789

211/387 [===============>..............] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0785

232/387 [================>.............] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0785

253/387 [==================>...........] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0785

273/387 [====================>.........] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0790

289/387 [=====================>........] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0790

308/387 [======================>.......] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0791

323/387 [========================>.....] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0792

341/387 [=========================>....] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0791

361/387 [==========================>...] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0791

380/387 [============================>.] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0791

387/387 [==============================] - 1s 3ms/step - loss: 0.0107 - mean_absolute_error: 0.0789


Epoch 11/150


  1/387 [..............................] - ETA: 1s - loss: 0.0119 - mean_absolute_error: 0.0951

 22/387 [>.............................] - ETA: 0s - loss: 0.0097 - mean_absolute_error: 0.0769

 41/387 [==>...........................] - ETA: 0s - loss: 0.0102 - mean_absolute_error: 0.0760

 62/387 [===>..........................] - ETA: 0s - loss: 0.0100 - mean_absolute_error: 0.0759

 80/387 [=====>........................] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0772

100/387 [======>.......................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0774

117/387 [========>.....................] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0793

132/387 [=========>....................] - ETA: 0s - loss: 0.0112 - mean_absolute_error: 0.0794

153/387 [==========>...................] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0798

171/387 [============>.................] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0799

192/387 [=============>................] - ETA: 0s - loss: 0.0112 - mean_absolute_error: 0.0798

210/387 [===============>..............] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0791

226/387 [================>.............] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0789

248/387 [==================>...........] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0791

274/387 [====================>.........] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0784

299/387 [======================>.......] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0786

324/387 [========================>.....] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0783

338/387 [=========================>....] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0783

363/387 [===========================>..] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0784

387/387 [==============================] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0784

387/387 [==============================] - 1s 3ms/step - loss: 0.0106 - mean_absolute_error: 0.0784


Epoch 12/150


  1/387 [..............................] - ETA: 1s - loss: 0.0053 - mean_absolute_error: 0.0601

 25/387 [>.............................] - ETA: 0s - loss: 0.0093 - mean_absolute_error: 0.0765

 51/387 [==>...........................] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0785

 72/387 [====>.........................] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0805

 96/387 [======>.......................] - ETA: 0s - loss: 0.0112 - mean_absolute_error: 0.0805

122/387 [========>.....................] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0794

147/387 [==========>...................] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0796

173/387 [============>.................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0784

193/387 [=============>................] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0783

217/387 [===============>..............] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0787

243/387 [=================>............] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0786

263/387 [===================>..........] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0785

288/387 [=====================>........] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0780

307/387 [======================>.......] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0781

329/387 [========================>.....] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0779

353/387 [==========================>...] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0779

376/387 [============================>.] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0787

387/387 [==============================] - 1s 2ms/step - loss: 0.0106 - mean_absolute_error: 0.0787


Epoch 13/150


  1/387 [..............................] - ETA: 1s - loss: 0.0145 - mean_absolute_error: 0.0948

 24/387 [>.............................] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0779

 41/387 [==>...........................] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0792

 68/387 [====>.........................] - ETA: 0s - loss: 0.0102 - mean_absolute_error: 0.0776

 93/387 [======>.......................] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0802

117/387 [========>.....................] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0807

143/387 [==========>...................] - ETA: 0s - loss: 0.0111 - mean_absolute_error: 0.0799

164/387 [===========>..................] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0787

186/387 [=============>................] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0787

208/387 [===============>..............] - ETA: 0s - loss: 0.0111 - mean_absolute_error: 0.0792

226/387 [================>.............] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0790

248/387 [==================>...........] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0793

260/387 [===================>..........] - ETA: 0s - loss: 0.0112 - mean_absolute_error: 0.0799

279/387 [====================>.........] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0796

297/387 [======================>.......] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0793

322/387 [=======================>......] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0791

341/387 [=========================>....] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0787

357/387 [==========================>...] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0788

375/387 [============================>.] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0787

387/387 [==============================] - 1s 3ms/step - loss: 0.0106 - mean_absolute_error: 0.0788


Epoch 14/150


  1/387 [..............................] - ETA: 1s - loss: 0.0079 - mean_absolute_error: 0.0702

 19/387 [>.............................] - ETA: 1s - loss: 0.0101 - mean_absolute_error: 0.0784

 42/387 [==>...........................] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0784

 56/387 [===>..........................] - ETA: 0s - loss: 0.0100 - mean_absolute_error: 0.0774

 71/387 [====>.........................] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0790

 91/387 [======>.......................] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0785

113/387 [=======>......................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0781

132/387 [=========>....................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0784

150/387 [==========>...................] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0785

169/387 [============>.................] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0791

189/387 [=============>................] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0790

210/387 [===============>..............] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0787

231/387 [================>.............] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0798

251/387 [==================>...........] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0799

264/387 [===================>..........] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0795

284/387 [=====================>........] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0798

302/387 [======================>.......] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0793

320/387 [=======================>......] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0794

341/387 [=========================>....] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0789

359/387 [==========================>...] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0788

380/387 [============================>.] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0788

387/387 [==============================] - 1s 3ms/step - loss: 0.0106 - mean_absolute_error: 0.0787


Epoch 15/150


  1/387 [..............................] - ETA: 2s - loss: 0.0089 - mean_absolute_error: 0.0694

 18/387 [>.............................] - ETA: 1s - loss: 0.0110 - mean_absolute_error: 0.0852

 39/387 [==>...........................] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0812

 59/387 [===>..........................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0789

 77/387 [====>.........................] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0787

101/387 [======>.......................] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0792

120/387 [========>.....................] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0789

138/387 [=========>....................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0783

155/387 [===========>..................] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0784

175/387 [============>.................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0779

199/387 [==============>...............] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0779

221/387 [================>.............] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0781

245/387 [=================>............] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0781

268/387 [===================>..........] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0777

286/387 [=====================>........] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0781

306/387 [======================>.......] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0780

323/387 [========================>.....] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0783

343/387 [=========================>....] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0786

368/387 [===========================>..] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0780

387/387 [==============================] - 1s 3ms/step - loss: 0.0105 - mean_absolute_error: 0.0781


Epoch 16/150


  1/387 [..............................] - ETA: 1s - loss: 0.0052 - mean_absolute_error: 0.0634

 28/387 [=>............................] - ETA: 0s - loss: 0.0117 - mean_absolute_error: 0.0783

 51/387 [==>...........................] - ETA: 0s - loss: 0.0112 - mean_absolute_error: 0.0795

 73/387 [====>.........................] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0797

 89/387 [=====>........................] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0800

 98/387 [======>.......................] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0803

117/387 [========>.....................] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0792

140/387 [=========>....................] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0795

162/387 [===========>..................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0790

182/387 [=============>................] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0786

198/387 [==============>...............] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0784

217/387 [===============>..............] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0787

243/387 [=================>............] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0789

266/387 [===================>..........] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0788

284/387 [=====================>........] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0786

300/387 [======================>.......] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0781

326/387 [========================>.....] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0778

352/387 [==========================>...] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0780

371/387 [===========================>..] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0784

387/387 [==============================] - 1s 3ms/step - loss: 0.0105 - mean_absolute_error: 0.0783


Epoch 17/150


  1/387 [..............................] - ETA: 2s - loss: 0.0188 - mean_absolute_error: 0.1056

 18/387 [>.............................] - ETA: 1s - loss: 0.0090 - mean_absolute_error: 0.0744

 42/387 [==>...........................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0771

 66/387 [====>.........................] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0797

 89/387 [=====>........................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0791

107/387 [=======>......................] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0776

124/387 [========>.....................] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0777

137/387 [=========>....................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0775

160/387 [===========>..................] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0779

182/387 [=============>................] - ETA: 0s - loss: 0.0102 - mean_absolute_error: 0.0773

208/387 [===============>..............] - ETA: 0s - loss: 0.0102 - mean_absolute_error: 0.0774

229/387 [================>.............] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0782

245/387 [=================>............] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0780

262/387 [===================>..........] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0781

280/387 [====================>.........] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0780

306/387 [======================>.......] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0782

329/387 [========================>.....] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0785

347/387 [=========================>....] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0784

369/387 [===========================>..] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0781

387/387 [==============================] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0782

387/387 [==============================] - 1s 3ms/step - loss: 0.0105 - mean_absolute_error: 0.0782


Epoch 18/150


  1/387 [..............................] - ETA: 1s - loss: 0.0067 - mean_absolute_error: 0.0622

 21/387 [>.............................] - ETA: 0s - loss: 0.0101 - mean_absolute_error: 0.0773

 39/387 [==>...........................] - ETA: 0s - loss: 0.0093 - mean_absolute_error: 0.0743

 64/387 [===>..........................] - ETA: 0s - loss: 0.0099 - mean_absolute_error: 0.0771

 85/387 [=====>........................] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0793

108/387 [=======>......................] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0785

127/387 [========>.....................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0780

146/387 [==========>...................] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0795

161/387 [===========>..................] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0794

181/387 [=============>................] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0791

201/387 [==============>...............] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0789

219/387 [===============>..............] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0782

239/387 [=================>............] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0779

259/387 [===================>..........] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0774

281/387 [====================>.........] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0776

300/387 [======================>.......] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0780

317/387 [=======================>......] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0780

335/387 [========================>.....] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0778

348/387 [=========================>....] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0778

367/387 [===========================>..] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0777

386/387 [============================>.] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0781

387/387 [==============================] - 1s 3ms/step - loss: 0.0105 - mean_absolute_error: 0.0781


Epoch 19/150


  1/387 [..............................] - ETA: 1s - loss: 0.0103 - mean_absolute_error: 0.0852

 21/387 [>.............................] - ETA: 0s - loss: 0.0129 - mean_absolute_error: 0.0860

 38/387 [=>............................] - ETA: 1s - loss: 0.0125 - mean_absolute_error: 0.0852

 56/387 [===>..........................] - ETA: 0s - loss: 0.0116 - mean_absolute_error: 0.0829

 78/387 [=====>........................] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0809

 99/387 [======>.......................] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0814

122/387 [========>.....................] - ETA: 0s - loss: 0.0112 - mean_absolute_error: 0.0816

139/387 [=========>....................] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0810

155/387 [===========>..................] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0809

176/387 [============>.................] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0804

196/387 [==============>...............] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0796

218/387 [===============>..............] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0793

237/387 [=================>............] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0794

255/387 [==================>...........] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0787

277/387 [====================>.........] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0785

299/387 [======================>.......] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0782

318/387 [=======================>......] - ETA: 0s - loss: 0.0102 - mean_absolute_error: 0.0780

339/387 [=========================>....] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0785

358/387 [==========================>...] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0787

374/387 [===========================>..] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0786

387/387 [==============================] - 1s 3ms/step - loss: 0.0106 - mean_absolute_error: 0.0790


Epoch 20/150


  1/387 [..............................] - ETA: 1s - loss: 0.0050 - mean_absolute_error: 0.0596

 20/387 [>.............................] - ETA: 0s - loss: 0.0115 - mean_absolute_error: 0.0819

 41/387 [==>...........................] - ETA: 0s - loss: 0.0121 - mean_absolute_error: 0.0827

 62/387 [===>..........................] - ETA: 0s - loss: 0.0117 - mean_absolute_error: 0.0816

 82/387 [=====>........................] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0793

102/387 [======>.......................] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0789

123/387 [========>.....................] - ETA: 0s - loss: 0.0111 - mean_absolute_error: 0.0787

143/387 [==========>...................] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0787

166/387 [===========>..................] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0790

177/387 [============>.................] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0790

199/387 [==============>...............] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0787

220/387 [================>.............] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0786

239/387 [=================>............] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0784

260/387 [===================>..........] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0790

279/387 [====================>.........] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0786

299/387 [======================>.......] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0786

321/387 [=======================>......] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0783

341/387 [=========================>....] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0780

363/387 [===========================>..] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0778

380/387 [============================>.] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0779

387/387 [==============================] - 1s 3ms/step - loss: 0.0104 - mean_absolute_error: 0.0779


Epoch 21/150


  1/387 [..............................] - ETA: 2s - loss: 0.0100 - mean_absolute_error: 0.0894

 25/387 [>.............................] - ETA: 0s - loss: 0.0095 - mean_absolute_error: 0.0770

 47/387 [==>...........................] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0801

 68/387 [====>.........................] - ETA: 0s - loss: 0.0115 - mean_absolute_error: 0.0819

 89/387 [=====>........................] - ETA: 0s - loss: 0.0119 - mean_absolute_error: 0.0826

107/387 [=======>......................] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0809

126/387 [========>.....................] - ETA: 0s - loss: 0.0114 - mean_absolute_error: 0.0812

149/387 [==========>...................] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0813

171/387 [============>.................] - ETA: 0s - loss: 0.0111 - mean_absolute_error: 0.0805

191/387 [=============>................] - ETA: 0s - loss: 0.0111 - mean_absolute_error: 0.0803

206/387 [==============>...............] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0802

227/387 [================>.............] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0800

245/387 [=================>............] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0797

267/387 [===================>..........] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0791

287/387 [=====================>........] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0794

306/387 [======================>.......] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0791

329/387 [========================>.....] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0790

349/387 [==========================>...] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0786

368/387 [===========================>..] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0785

387/387 [==============================] - 1s 3ms/step - loss: 0.0105 - mean_absolute_error: 0.0785


Epoch 22/150


  1/387 [..............................] - ETA: 2s - loss: 0.0087 - mean_absolute_error: 0.0834

 16/387 [>.............................] - ETA: 1s - loss: 0.0101 - mean_absolute_error: 0.0776

 38/387 [=>............................] - ETA: 1s - loss: 0.0106 - mean_absolute_error: 0.0788

 55/387 [===>..........................] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0800

 76/387 [====>.........................] - ETA: 0s - loss: 0.0101 - mean_absolute_error: 0.0775

100/387 [======>.......................] - ETA: 0s - loss: 0.0101 - mean_absolute_error: 0.0773

118/387 [========>.....................] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0781

140/387 [=========>....................] - ETA: 0s - loss: 0.0102 - mean_absolute_error: 0.0776

162/387 [===========>..................] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0786

183/387 [=============>................] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0784

203/387 [==============>...............] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0786

221/387 [================>.............] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0791

236/387 [=================>............] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0791

252/387 [==================>...........] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0787

272/387 [====================>.........] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0786

292/387 [=====================>........] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0782

311/387 [=======================>......] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0784

329/387 [========================>.....] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0789

349/387 [==========================>...] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0784

368/387 [===========================>..] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0785

387/387 [==============================] - 1s 3ms/step - loss: 0.0106 - mean_absolute_error: 0.0785


Epoch 23/150


  1/387 [..............................] - ETA: 1s - loss: 0.0087 - mean_absolute_error: 0.0837

 16/387 [>.............................] - ETA: 1s - loss: 0.0109 - mean_absolute_error: 0.0802

 30/387 [=>............................] - ETA: 1s - loss: 0.0123 - mean_absolute_error: 0.0814

 49/387 [==>...........................] - ETA: 1s - loss: 0.0110 - mean_absolute_error: 0.0793

 70/387 [====>.........................] - ETA: 0s - loss: 0.0112 - mean_absolute_error: 0.0802

 91/387 [======>.......................] - ETA: 0s - loss: 0.0102 - mean_absolute_error: 0.0775

111/387 [=======>......................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0788

130/387 [=========>....................] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0794

151/387 [==========>...................] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0796

173/387 [============>.................] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0795

193/387 [=============>................] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0794

212/387 [===============>..............] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0792

227/387 [================>.............] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0787

248/387 [==================>...........] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0781

265/387 [===================>..........] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0784

285/387 [=====================>........] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0780

309/387 [======================>.......] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0787

329/387 [========================>.....] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0786

352/387 [==========================>...] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0784

374/387 [===========================>..] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0782

387/387 [==============================] - 1s 3ms/step - loss: 0.0105 - mean_absolute_error: 0.0783


Epoch 24/150


  1/387 [..............................] - ETA: 1s - loss: 0.0083 - mean_absolute_error: 0.0729

 25/387 [>.............................] - ETA: 0s - loss: 0.0097 - mean_absolute_error: 0.0755

 46/387 [==>...........................] - ETA: 0s - loss: 0.0099 - mean_absolute_error: 0.0761

 63/387 [===>..........................] - ETA: 0s - loss: 0.0094 - mean_absolute_error: 0.0756

 82/387 [=====>........................] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0782

102/387 [======>.......................] - ETA: 0s - loss: 0.0118 - mean_absolute_error: 0.0809

124/387 [========>.....................] - ETA: 0s - loss: 0.0114 - mean_absolute_error: 0.0803

144/387 [==========>...................] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0791

162/387 [===========>..................] - ETA: 0s - loss: 0.0112 - mean_absolute_error: 0.0797

181/387 [=============>................] - ETA: 0s - loss: 0.0111 - mean_absolute_error: 0.0792

199/387 [==============>...............] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0792

219/387 [===============>..............] - ETA: 0s - loss: 0.0111 - mean_absolute_error: 0.0796

231/387 [================>.............] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0792

248/387 [==================>...........] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0785

271/387 [====================>.........] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0781

288/387 [=====================>........] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0780

307/387 [======================>.......] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0781

327/387 [========================>.....] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0782

351/387 [==========================>...] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0785

374/387 [===========================>..] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0783

387/387 [==============================] - 1s 3ms/step - loss: 0.0105 - mean_absolute_error: 0.0781


Epoch 25/150


  1/387 [..............................] - ETA: 1s - loss: 0.0176 - mean_absolute_error: 0.1023

 24/387 [>.............................] - ETA: 0s - loss: 0.0092 - mean_absolute_error: 0.0754

 46/387 [==>...........................] - ETA: 0s - loss: 0.0085 - mean_absolute_error: 0.0734

 67/387 [====>.........................] - ETA: 0s - loss: 0.0089 - mean_absolute_error: 0.0749

 89/387 [=====>........................] - ETA: 0s - loss: 0.0094 - mean_absolute_error: 0.0771

117/387 [========>.....................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0799

144/387 [==========>...................] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0803

169/387 [============>.................] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0798

190/387 [=============>................] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0794

215/387 [===============>..............] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0794

243/387 [=================>............] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0785

268/387 [===================>..........] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0784

293/387 [=====================>........] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0789

314/387 [=======================>......] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0785

334/387 [========================>.....] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0784

361/387 [==========================>...] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0781

387/387 [==============================] - 1s 2ms/step - loss: 0.0105 - mean_absolute_error: 0.0782


Epoch 26/150


  1/387 [..............................] - ETA: 1s - loss: 0.0034 - mean_absolute_error: 0.0503

 28/387 [=>............................] - ETA: 0s - loss: 0.0085 - mean_absolute_error: 0.0750

 53/387 [===>..........................] - ETA: 0s - loss: 0.0089 - mean_absolute_error: 0.0749

 78/387 [=====>........................] - ETA: 0s - loss: 0.0092 - mean_absolute_error: 0.0748

105/387 [=======>......................] - ETA: 0s - loss: 0.0097 - mean_absolute_error: 0.0757

131/387 [=========>....................] - ETA: 0s - loss: 0.0101 - mean_absolute_error: 0.0770

155/387 [===========>..................] - ETA: 0s - loss: 0.0101 - mean_absolute_error: 0.0774

181/387 [=============>................] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0776

196/387 [==============>...............] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0777

224/387 [================>.............] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0784

249/387 [==================>...........] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0780

273/387 [====================>.........] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0780

294/387 [=====================>........] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0781

310/387 [=======================>......] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0782

333/387 [========================>.....] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0783

358/387 [==========================>...] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0780

383/387 [============================>.] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0784

387/387 [==============================] - 1s 2ms/step - loss: 0.0105 - mean_absolute_error: 0.0782


Epoch 27/150


  1/387 [..............................] - ETA: 1s - loss: 0.0047 - mean_absolute_error: 0.0560

 27/387 [=>............................] - ETA: 0s - loss: 0.0110 - mean_absolute_error: 0.0765

 45/387 [==>...........................] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0778

 71/387 [====>.........................] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0796

 95/387 [======>.......................] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0773

121/387 [========>.....................] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0779

149/387 [==========>...................] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0781

175/387 [============>.................] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0778

202/387 [==============>...............] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0788

229/387 [================>.............] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0787

253/387 [==================>...........] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0787

279/387 [====================>.........] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0782

295/387 [=====================>........] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0782

321/387 [=======================>......] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0783

346/387 [=========================>....] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0782

371/387 [===========================>..] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0782

387/387 [==============================] - 1s 2ms/step - loss: 0.0104 - mean_absolute_error: 0.0779


Epoch 28/150


  1/387 [..............................] - ETA: 1s - loss: 0.0102 - mean_absolute_error: 0.0827

 25/387 [>.............................] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0846

 46/387 [==>...........................] - ETA: 0s - loss: 0.0100 - mean_absolute_error: 0.0782

 71/387 [====>.........................] - ETA: 0s - loss: 0.0096 - mean_absolute_error: 0.0772

 96/387 [======>.......................] - ETA: 0s - loss: 0.0093 - mean_absolute_error: 0.0759

124/387 [========>.....................] - ETA: 0s - loss: 0.0091 - mean_absolute_error: 0.0757

146/387 [==========>...................] - ETA: 0s - loss: 0.0094 - mean_absolute_error: 0.0763

165/387 [===========>..................] - ETA: 0s - loss: 0.0098 - mean_absolute_error: 0.0773

191/387 [=============>................] - ETA: 0s - loss: 0.0101 - mean_absolute_error: 0.0784

214/387 [===============>..............] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0785

241/387 [=================>............] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0795

268/387 [===================>..........] - ETA: 0s - loss: 0.0106 - mean_absolute_error: 0.0791

290/387 [=====================>........] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0785

312/387 [=======================>......] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0782

332/387 [========================>.....] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0784

355/387 [==========================>...] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0782

376/387 [============================>.] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0781

387/387 [==============================] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0782

387/387 [==============================] - 1s 2ms/step - loss: 0.0104 - mean_absolute_error: 0.0782


Epoch 29/150


  1/387 [..............................] - ETA: 1s - loss: 0.0044 - mean_absolute_error: 0.0531

 20/387 [>.............................] - ETA: 1s - loss: 0.0095 - mean_absolute_error: 0.0737

 40/387 [==>...........................] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0786

 50/387 [==>...........................] - ETA: 1s - loss: 0.0104 - mean_absolute_error: 0.0794

 70/387 [====>.........................] - ETA: 1s - loss: 0.0108 - mean_absolute_error: 0.0814

 90/387 [=====>........................] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0818

104/387 [=======>......................] - ETA: 0s - loss: 0.0115 - mean_absolute_error: 0.0818

127/387 [========>.....................] - ETA: 0s - loss: 0.0113 - mean_absolute_error: 0.0815

153/387 [==========>...................] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0805

174/387 [============>.................] - ETA: 0s - loss: 0.0108 - mean_absolute_error: 0.0802

189/387 [=============>................] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0801

209/387 [===============>..............] - ETA: 0s - loss: 0.0109 - mean_absolute_error: 0.0804

226/387 [================>.............] - ETA: 0s - loss: 0.0107 - mean_absolute_error: 0.0799

245/387 [=================>............] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0788

267/387 [===================>..........] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0786

287/387 [=====================>........] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0788

311/387 [=======================>......] - ETA: 0s - loss: 0.0105 - mean_absolute_error: 0.0787

335/387 [========================>.....] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0784

357/387 [==========================>...] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0780

378/387 [============================>.] - ETA: 0s - loss: 0.0104 - mean_absolute_error: 0.0781

387/387 [==============================] - 1s 3ms/step - loss: 0.0104 - mean_absolute_error: 0.0779


Epoch 30/150


  1/387 [..............................] - ETA: 1s - loss: 0.0163 - mean_absolute_error: 0.0946

 20/387 [>.............................] - ETA: 1s - loss: 0.0097 - mean_absolute_error: 0.0756

 40/387 [==>...........................] - ETA: 0s - loss: 0.0102 - mean_absolute_error: 0.0775

 57/387 [===>..........................] - ETA: 0s - loss: 0.0099 - mean_absolute_error: 0.0762

 76/387 [====>.........................] - ETA: 0s - loss: 0.0096 - mean_absolute_error: 0.0755

 95/387 [======>.......................] - ETA: 0s - loss: 0.0095 - mean_absolute_error: 0.0747

118/387 [========>.....................] - ETA: 0s - loss: 0.0095 - mean_absolute_error: 0.0748

142/387 [==========>...................] - ETA: 0s - loss: 0.0099 - mean_absolute_error: 0.0763

160/387 [===========>..................] - ETA: 0s - loss: 0.0100 - mean_absolute_error: 0.0769

179/387 [============>.................] - ETA: 0s - loss: 0.0100 - mean_absolute_error: 0.0772

194/387 [==============>...............] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0778

215/387 [===============>..............] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0777

236/387 [=================>............] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0777

258/387 [===================>..........] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0775

281/387 [====================>.........] - ETA: 0s - loss: 0.0103 - mean_absolute_error: 0.0774

In [18]:
model.metrics_names

['loss', 'mean_absolute_error']

In [19]:
model.evaluate(model_X_basic_test, Y_test, verbose = 0)

[0.01021327544003725, 0.0775848850607872]

In [20]:
pred_nn = model.predict(model_X_basic_test)
pred_nn

41/41 [==============================] - 0s 1ms/step


array([[0.31040493],
       [0.42235863],
       [0.39654243],
       ...,
       [0.3823735 ],
       [0.40649807],
       [0.38370064]], dtype=float32)

In [21]:
import statsmodels.api as sm
import statsmodels.formula.api as smf

In [22]:
resid_basic = (Y_test-pred_nn)**2

In [23]:
MSE_nn_basic = sm.OLS( resid_basic , np.ones( resid_basic.shape[0] ) ).fit().summary2().tables[1].iloc[0, 0:2]
MSE_nn_basic

Coef.       0.010213
Std.Err.    0.000552
Name: const, dtype: float64

In [24]:
R2_nn_basic = 1 - ( MSE_nn_basic[0]/Y_test.var() )
print( f"The R^2 using NN is equal to = {R2_nn_basic[0]}" ) # MSE NN (basic model) 

The R^2 using NN is equal to = 0.1511966373372956
